docs/admin/federation/kubefed.md

```diff
@@ -0,0 +1,168 @@
+kubefed init fellowship --host-cluster-context=rivendell  --dns-zone-name="example.com"
+```
+
+where `fellowship` is the name of the federation, `rivendell` is the
+host cluster context from the previous step and `example.com` is the
+domain name suffix that you want for your federated services. This has
+to be a real domain that you control over and is programmable by your
+DNS provider.
+
+`kubefed init` sets up the federation control plane in the host
+cluster and also adds an entry for the federation API server in your
+local kubeconfig. However, in this alpha release it does not
+automatically set the current context to the newly deployed federation.
+You can set this by running:
+
+```shell
+kubectl config use-context fellowship
+```
+
+where `fellowship` is the name of your federation.
+
+## Joining a cluster
+
+Cluster Federation allows you to manage your workload across multiple
+clusters. In order to do that, a federation control plane must be
+aware of the clusters that it is responsible for managing. You can
+teach the federation control plane about a cluster by joining that
+cluster to the federation.
+
+Joining a cluster to a federation is as simple as running:
+
+```
+kubfed join gondor --host-cluster-context=rivendell
+```
+
+where `gondor` is the name of the cluster you are joining and
+`rivendell` is the federation control plane's host cluster.
+
+### Naming rules and customization
+
+#### Cluster name
+
+Cluster name supplied to `kubefed join` must be a valid RFC 1035 label.
+
+Furthermore, federation control plane requires credentials of the
+joined clusters to operate on them. These credentials are obtained
+from the local kubeconfig. `kubefed join` uses the cluster name
+specified as the argument to look for the cluster's context in the
+local kubeconfig. If it fails to find a matching context, it exits
+with an error.
+
+It is particularly a problem in the cases where context names
+for the clusters don't follow RFC 1035 label naming rules. In these
+cases, you could specify a cluster name that conforms to the RFC 1035
+label naming rules and specify the cluster context using the
+`--cluster-context` flag. For example, if context of the cluster your
+are joining is `gondor_has_no_king-gondor_needs_no_king`, then you can
+join the cluster by running:
+
+```shell
+kubfed join gondor --host-cluster-context=rivendell --cluster-context=gondor_has_no_king-gondor_needs_no_king
+```
+
+#### Secret name
+
+Cluster credentials required by the federation control plane as described above are stored as a secret in the host cluster. The name of the secret is also derived from the cluster name. 
+
+However, the name of a secret object in Kubernetes should conform
+to the subdomain name specification described in RFC 1123. If this isn't case, you can pass the secret name to `kubefed join` using the
+`--secret-name` flag. For example, if the cluster name is `noldor` and
+the secret name is `11kingdom`, you can join the cluster by running:
```

Comment:
That won't demonstrate a valid secret name but an invalid cluster name. The difference are the numbers in the front.